### PR TITLE
[ci-skip] Creation of vault auth from vault-k8s chart.

### LINKED ISSUE #2289: feat(shared): creation of vault auth from shared vault-k8s chart
### Description
As a developer I want to have the vault auth service account and  k8s rbac.authorization clusterrolebinding created from vault-k8s charts.

- [x] HL Besu
- [x] Quorum
- [x] Corda OS
### Acceptance Criteria
1. Test for each platform using shared vault-k8s chart
2. Remove the ansible reference for vault reviewer

### PR DESCRIPTION
**Pull request by mgCepeda**
Primary Changes
--------------
1. Delete creation vault auth service account and k8s rbac.authorization clusterrolebinding from ansible
2. Delete templates for creating vault-auth, vault-reviewer and clusterrolebinding
3. Add creation vault auth service account and k8s rbac.authorization clusterrolebinding from vault-k8s charts.

Modifications in /hyperledger-besu
platforms/hyperledger-besu/configuration/add-validator.yaml 
platforms/hyperledger-besu/configuration/deploy-network.yaml 
platforms/hyperledger-besu/configuration/generate-crypto.yaml 
platforms/hyperledger-besu/configuration/roles/create/k8_component/vars/main.yaml 
platforms/hyperledger-besu/configuration/roles/create/namespace_serviceaccount/tasks/main.yaml

Modifications in quorum
platforms/quorum/configuration/deploy-network.yaml platforms/quorum/configuration/roles/create/k8_component/vars/main.yaml platforms/quorum/configuration/roles/create/namespace_serviceaccount/tasks/main.yaml platforms/quorum/releases/README.md

Modifications in r3-corda
platforms/r3-corda/configuration/deploy-network.yaml 
platforms/r3-corda/configuration/roles/create/k8_component/tasks/main.yaml 
platforms/r3-corda/configuration/roles/create/k8_component/vars/main.yaml 
platforms/r3-corda/configuration/roles/create/namespace_serviceaccount/tasks/main.yaml 
platforms/r3-corda/configuration/roles/setup/doorman/tasks/main.yml 
platforms/r3-corda/configuration/roles/setup/nms/tasks/main.yaml 
platforms/r3-corda/configuration/roles/setup/node/tasks/main.yaml 
platforms/r3-corda/configuration/roles/setup/notary/tasks/main.yaml

Modifications in shared
platforms/shared/charts/vault-k8s-mgmt/templates/job.yaml platforms/shared/charts/vault-k8s-mgmt/values.yaml platforms/shared/configuration/roles/create/shared_helm_component/templates/vault_kubernetes_job.tpl platforms/shared/configuration/roles/setup/vault_kubernetes/tasks/main.yaml

Fixes
#2289

### CODE REVIEW

File: platforms/shared/charts/vault-k8s-mgmt/templates/job.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ spec:
       restartPolicy: OnFailure
       imagePullSecrets:
         - name: {{ $.Values.vault.imagesecretname }}
-      serviceAccountName: {{ $.Values.vault.reviewer_service }}
+      serviceAccountName: {{ $.Values.serviceAccount.name }}
```
Comment (sownak) [inline platforms/shared/charts/vault-k8s-mgmt/templates/job.yaml:33]: As vault-auth is created by the same chart, 'vault-auth' can be hardcoded here. Or update the templates/serviceAccount.yaml to use the Helm values to name serviceaccount

File: platforms/shared/charts/vault-k8s-mgmt/templates/serviceAccount.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{{- if .Values.serviceAccount.create -}}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: vault-auth
```
Comment (sownak) [inline platforms/shared/charts/vault-k8s-mgmt/templates/serviceAccount.yaml:5]: Should be read from helm values as I mentioned above

File: platforms/shared/charts/vault-k8s-mgmt/values.yaml [hunk 3]
```diff
@@ -58,3 +52,14 @@ k8s:
...
+
+rbac:
+  # Specifies whether RBAC resources should be created
+  create: true
+
+serviceAccount:
+  # Specifies whether a ServiceAccount should be created
+  create: true
+  # The name of the ServiceAccount to use.
+  # If not set and create is true, a name is generated using the fullname template
```
Comment (sownak) [inline platforms/shared/charts/vault-k8s-mgmt/values.yaml:64]: Update the chart.yaml with new version and Readme

**Review by sownak — CHANGES_REQUESTED** [review]

File: platforms/shared/charts/vault-k8s-mgmt/templates/job.yaml (hunk 1 above)
Comment (mgCepeda) [reply platforms/shared/charts/vault-k8s-mgmt/templates/job.yaml:33]: Done

File: platforms/shared/charts/vault-k8s-mgmt/templates/serviceAccount.yaml (hunk 2 above)
Comment (mgCepeda) [reply platforms/shared/charts/vault-k8s-mgmt/templates/serviceAccount.yaml]: Done

File: platforms/shared/charts/vault-k8s-mgmt/values.yaml (hunk 3 above)
Comment (mgCepeda) [reply platforms/shared/charts/vault-k8s-mgmt/values.yaml:64]: Done

File: platforms/shared/charts/vault-k8s-mgmt/templates/serviceAccount.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+metadata:
+  name: {{ .Values.metadata.namespace }}-role-tokenreview-binding
+  namespace: default
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:auth-delegator
+subjects:
+- kind: ServiceAccount
+  name: vault-auth
```
Comment (sownak) [inline platforms/shared/charts/vault-k8s-mgmt/templates/serviceAccount.yaml:21]: Parameter is missed here

Comment (mgCepeda) [reply platforms/shared/charts/vault-k8s-mgmt/templates/serviceAccount.yaml]: Done

**Review by sownak — APPROVED** [review]

**Pull request merged**